src/Microsoft.DotNet.Build.CloudTestTasks/SendToHelix.cs

```diff
@@ -63,53 +63,58 @@ public override bool Execute()
 
                 while (true)
                 {
-                    HttpResponseMessage response;
-                                        
-                    using (Stream stream = File.OpenRead(EventDataPath))
-                    {
-                        HttpContent contentStream = new StreamContent(stream);
-                        contentStream.Headers.Add("Content-Type", "application/json");
-                        response = await client.PostAsync(apiUrl, contentStream);
-                    }
+                    HttpResponseMessage response = new HttpResponseMessage();
 
-                    if (response.IsSuccessStatusCode)
+                    try
                     {
-                        JObject responseObject;
-                        using (Stream stream = await response.Content.ReadAsStreamAsync())
-                        using (StreamReader streamReader = new StreamReader(stream))
-                        using (JsonReader jsonReader = new JsonTextReader(streamReader))
+                        using (Stream stream = File.OpenRead(EventDataPath))
                         {
-                            responseObject = JObject.Load(jsonReader);
+                            HttpContent contentStream = new StreamContent(stream);
+                            contentStream.Headers.Add("Content-Type", "application/json");
+                            response = await client.PostAsync(apiUrl, contentStream);
```

Comment:
It looks like this wasn't getting disposed before, so we were probably getting lucky (and things got finalized when the task exited).  I know there are dangers of deadlocking when not disposing networking related resources. But it looks like this code isn't introducing new problems (just the diff on code flow makes it look like it is)